src/time_integrator.f90

```diff
@@ -167,33 +159,12 @@ function init(backend, allocator, method, nvars)
 
   end function init
 
-  subroutine step(self, u, v, w, du, dv, dw, dt)
-    implicit none
-
-    class(time_intg_t), intent(inout) :: self
-    class(field_t), target, intent(inout) :: u, v, w
-    class(field_t), target, intent(in) :: du, dv, dw
-
-    real(dp), intent(in) :: dt
-
-    ! assign pointer to variables
-    self%curr(1)%ptr => u
-    self%curr(2)%ptr => v
-    self%curr(3)%ptr => w
-
-    ! assign pointer to variables
-    self%deriv(1)%ptr => du
-    self%deriv(2)%ptr => dv
-    self%deriv(3)%ptr => dw
-
-    call self%stepper(dt)
-
-  end subroutine step
-
-  subroutine runge_kutta(self, dt)
+  subroutine runge_kutta(self, curr, deriv, dt)
     implicit none
 
     class(time_intg_t), intent(inout) :: self
+    type(flist_t), allocatable, intent(inout) :: curr(:)
+    type(flist_t), allocatable, intent(in) :: deriv(:)
```

Comment:
If an object is allocated/deallocated outside we typically don't include `allocatable` attribute. Having `allocatable` attribute midly implies we may alloc/dealloc inside the subroutine, which is not the case here. Probably not going to interfere with the behaviour but I think its better to omit it here.